src/components/calcite-block-section/calcite-block-section.tsx

```diff
@@ -68,28 +79,44 @@ export class CalciteBlockSection {
   //
   // --------------------------------------------------------------------------
 
-  onHeaderClick = () => {
+  toggleSection = () => {
     this.open = !this.open;
     this.calciteBlockSectionToggle.emit();
   };
 
+  handleSwitchClick = ({ currentTarget, target }: MouseEvent) => {
+    const isDupeClick = target === currentTarget;
```

Comment:
Is this because the whole block is clickable? Should we maybe not have the area around the switch be a hit area for the block?

Reply:
It's caused by the wrapping label. It fires two clicks when you click on the switch. I'm going to open an issue on `calcite-components` about maybe stopping propagation or default behavior when there will be two clicks (it already has some code to check if its wrapped in a label).

Reply:
boulder: Swap from using an `onClick` event to an `onChange` event, and this problem goes away.

Reply:
I did this originally, but had some weird behavior. I'll revisit.

Reply:
I tried it out locally and it seemed to work fine - if you run into it again and get stuck, let me know and I'll help find a solution.

Reply:
Fixed. My weirdness came from using `onCalciteSwitchChange`. 🙇